### PR TITLE
*: update tikv/client-go to fix batch-client send loop panic issue

### LINKED ISSUE #47691: client-go batch client batchSendLoop panic
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

ref: https://github.com/tikv/client-go/issues/1020

Panic stack log:
```
[2023/10/13 14:00:33.948 +08:00] [ERROR] [client_batch.go:303] [batchSendLoop] [r={}] [stack="[github.com/tikv/client-go/v2/internal/client.(*batchConn).batchSendLoop.func1](http://github.com/tikv/client-go/v2/internal/client.(*batchConn).batchSendLoop.func1)
	/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.4-0.20230912041415-9c163cc8574b/internal/client/client_batch.go:305](http://github.com/tikv/client-go/v2@v2.0.4-0.20230912041415-9c163cc8574b/internal/client/client_batch.go:305)
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:884
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:113
[github.com/pingcap/kvproto/pkg/tikvpb.encodeVarintTikvpb](http://github.com/pingcap/kvproto/pkg/tikvpb.encodeVarintTikvpb)
	/go/pkg/mod/[github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:5438](http://github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:5438)
[github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest_Request_Coprocessor).MarshalToSizedBuffer](http://github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest_Request_Coprocessor).MarshalToSizedBuffer)
	/go/pkg/mod/[github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:4325](http://github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:4325)
[github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest_Request_Coprocessor).MarshalTo](http://github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest_Request_Coprocessor).MarshalTo)
	/go/pkg/mod/[github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:4313](http://github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:4313)
[github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest_Request).MarshalToSizedBuffer](http://github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest_Request).MarshalToSizedBuffer)
	/go/pkg/mod/[github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:3850](http://github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:3850)
[github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest).MarshalToSizedBuffer](http://github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest).MarshalToSizedBuffer)
	/go/pkg/mod/[github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:3808](http://github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:3808)
[github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest).Marshal](http://github.com/pingcap/kvproto/pkg/tikvpb.(*BatchCommandsRequest).Marshal)
	/go/pkg/mod/[github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:3766](http://github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:3766)
[google.golang.org/protobuf/internal/impl.legacyMarshal](http://google.golang.org/protobuf/internal/impl.legacyMarshal)
	/go/pkg/mod/[google.golang.org/protobuf@v1.28.1/internal/impl/legacy_message.go:402](http://google.golang.org/protobuf@v1.28.1/internal/impl/legacy_message.go:402)
[google.golang.org/protobuf/proto.MarshalOptions.marshal](http://google.golang.org/protobuf/proto.MarshalOptions.marshal)
	/go/pkg/mod/[google.golang.org/protobuf@v1.28.1/proto/encode.go:166](http://google.golang.org/protobuf@v1.28.1/proto/encode.go:166)
[google.golang.org/protobuf/proto.MarshalOptions.MarshalAppend](http://google.golang.org/protobuf/proto.MarshalOptions.MarshalAppend)
	/go/pkg/mod/[google.golang.org/protobuf@v1.28.1/proto/encode.go:125](http://google.golang.org/protobuf@v1.28.1/proto/encode.go:125)
[github.com/golang/protobuf/proto.marshalAppend](http://github.com/golang/protobuf/proto.marshalAppend)
	/go/pkg/mod/[github.com/golang/protobuf@v1.5.2/proto/wire.go:40](http://github.com/golang/protobuf@v1.5.2/proto/wire.go:40)
[github.com/golang/protobuf/proto.Marshal](http://github.com/golang/protobuf/proto.Marshal)
	/go/pkg/mod/[github.com/golang/protobuf@v1.5.2/proto/wire.go:23](http://github.com/golang/protobuf@v1.5.2/proto/wire.go:23)
[google.golang.org/grpc/encoding/proto.codec.Marshal](http://google.golang.org/grpc/encoding/proto.codec.Marshal)
	/go/pkg/mod/[google.golang.org/grpc@v1.51.0/encoding/proto/proto.go:45](http://google.golang.org/grpc@v1.51.0/encoding/proto/proto.go:45)
[google.golang.org/grpc.encode](http://google.golang.org/grpc.encode)
	/go/pkg/mod/[google.golang.org/grpc@v1.51.0/rpc_util.go:595](http://google.golang.org/grpc@v1.51.0/rpc_util.go:595)
[google.golang.org/grpc.prepareMsg](http://google.golang.org/grpc.prepareMsg)
	/go/pkg/mod/[google.golang.org/grpc@v1.51.0/stream.go:1708](http://google.golang.org/grpc@v1.51.0/stream.go:1708)
[google.golang.org/grpc.(*clientStream).SendMsg](http://google.golang.org/grpc.(*clientStream).SendMsg)
	/go/pkg/mod/[google.golang.org/grpc@v1.51.0/stream.go:846](http://google.golang.org/grpc@v1.51.0/stream.go:846)
[github.com/pingcap/kvproto/pkg/tikvpb.(*tikvBatchCommandsClient).Send](http://github.com/pingcap/kvproto/pkg/tikvpb.(*tikvBatchCommandsClient).Send)
	/go/pkg/mod/[github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:2068](http://github.com/pingcap/kvproto@v0.0.0-20230726063044-73d6d7f3756b/pkg/tikvpb/tikvpb.pb.go:2068)
[github.com/tikv/client-go/v2/internal/client.(*batchCommandsClient).send](http://github.com/tikv/client-go/v2/internal/client.(*batchCommandsClient).send)
	/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.4-0.20230912041415-9c163cc8574b/internal/client/client_batch.go:519](http://github.com/tikv/client-go/v2@v2.0.4-0.20230912041415-9c163cc8574b/internal/client/client_batch.go:519)
[github.com/tikv/client-go/v2/internal/client.(*batchConn).getClientAndSend](http://github.com/tikv/client-go/v2/internal/client.(*batchConn).getClientAndSend)
	/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.4-0.20230912041415-9c163cc8574b/internal/client/client_batch.go:381](http://github.com/tikv/client-go/v2@v2.0.4-0.20230912041415-9c163cc8574b/internal/client/client_batch.go:381)
[github.com/tikv/client-go/v2/internal/client.(*batchConn).batchSendLoop](http://github.com/tikv/client-go/v2/internal/client.(*batchConn).batchSendLoop)
	/go/pkg/mod/[github.com/tikv/client-go/v2@v2.0.4-0.20230912041415-9c163cc8574b/internal/client/client_batch.go:344](http://github.com/tikv/client-go/v2@v2.0.4-0.20230912041415-9c163cc8574b/internal/client/client_batch.go:344)"]
```

### 4. What is your TiDB version? (Required)

v6.5.5

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: close #47691

related PR: https://github.com/tikv/client-go/pull/1021

Problem Summary: update tikv/client-go to fix batch-client send loop panic issue

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
Side effects

- [x] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- N/A

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/realtikvtest/sessiontest/session_fail_test.go [hunk 1]
```diff
@@ -317,3 +320,41 @@ func TestTiKVClientReadTimeout(t *testing.T) {
...
+					// mock for kill query execution or timeout.
+					time.Sleep(time.Millisecond * time.Duration(rand.Intn(10)+1))
+					cancel()
+				}()
+				rs, _ := tk.ExecWithContext(ctx, "select * from t where a = 1")
+				if rs != nil {
+					session.ResultSetToStringSlice(ctx, tk.Session(), rs)
+				}
+			}
+
```
Comment (zimulala) [inline tests/realtikvtest/sessiontest/session_fail_test.go:356]: Please remove this useless line.
Besides, I ran this test（with `--with-real-tikv` and `-race`） locally with the master version 2 twice and passed both times. Is this test unreliable or does it need to work well on CI?

Comment (crazycs520) [reply tests/realtikvtest/sessiontest/session_fail_test.go]: Nice catch. I wrote this test in the release-6.5 branch first time, and the test will fail in release-6.5 branch.

Master branch can pass the test even without https://github.com/tikv/client-go/pull/1021 , I still keep the test to detect data race in future.

**Review by zimulala — APPROVED** [review]
LGTM

File: tests/realtikvtest/sessiontest/session_fail_test.go (hunk 1 above)
Comment (crazycs520) [reply tests/realtikvtest/sessiontest/session_fail_test.go]: In release-6.5 run this test will fail.
```sh
github.com/pingcap/tidb  release-6.5 ✗                                                                                                                                          1d23h ⚑
▶ cd tests/realtikvtest/sessiontest/
(base)
tests/realtikvtest/sessiontest  release-6.5 ✗                                                                                                                                   1d23h ⚑
▶ go test -race --tags=intest -run=TestBatchClientDataRace --with-real-tikv=true
...
...
...
[2023/10/18 14:56:15.607 +08:00] [INFO] [pd.go:209] ["closed pd http client"]
--- FAIL: TestBatchClientDataRace (31.54s)
    testing.go:1465: race detected during execution of test
FAIL
exit status 1
FAIL    github.com/pingcap/tidb/tests/realtikvtest/sessiontest  32.724s
```

**Review by nolouch — APPROVED** [review]

**Comment by cfzjywxk** [comment]
@crazycs520 
Is it already picked to the release-7.5 branch? If not we need to pick the fix to release-7.5 it's a blocking issue.

**Comment by crazycs520** [comment]
https://github.com/pingcap/tidb/pull/48109 already updated the go.mod.

**Pull request closed without merging**